runtime/vm/jvminit.c

```diff
@@ -6000,6 +6002,291 @@ processCompressionOptions(J9JavaVM *vm){
+									|| (NULL != upToColon)
+									|| (NULL != fullOptions)
+								) {
+									/* Invalid use of keyword disable */
+									goto xlogerr;
+								} else {
+									setgclog = TRUE;
+									gclog = FALSE;
+								}
+							} else if ((0 == j9_cmdla_stricmp("gc", tag)) || (0 == j9_cmdla_stricmp("gc*", tag))) {
+								checkTagLevel = TRUE;
+							} else {
+								unrecognizedOption = TRUE;
+							}
+							numtags += 1;
+							if (NULL == upToEquals) {
+								break;
+							}
+							tag = strUpToDelimiter(&upToEquals, '+');
+						}
+					}
+					if (checkTagLevel) {
+						/* look for a level after the '=' */
+						setgclog = TRUE;
+						if (NULL == upToComma) {
+							gclog = TRUE;
+						} else {
+							if ('\0' == *upToComma) {
+								goto xlogerr;
+							}
+							gclog = 0 != j9_cmdla_stricmp("off", upToComma);
+							if (gclog) {
+								/* OpenJ9 doesn't understand log levels */
+								unrecognizedOption = TRUE;
+							}
+						}
+					}
+					if (NULL == upToColon) {
+						break;
+					}
+					upToComma = strUpToDelimiter(&upToColon, ',');
+				}
+			}
+			if (NULL != fullOptions) {
+				/* look at the output method */
+				char *output = NULL;
+				upToColon = strUpToDelimiter(&fullOptions, ':');
+#if defined(DEBUG_XLOG)
+				j9tty_printf(PORTLIB, "upToColon %s fullOptions %s\n", upToColon, fullOptions);
+#endif /* defined(DEBUG_XLOG) */
+				output = strUpToDelimiter(&upToColon, '=');
+#if defined(DEBUG_XLOG)
+				j9tty_printf(PORTLIB, "output %s upToColon %s\n", output, upToColon);
+#endif /* defined(DEBUG_XLOG) */
+				/* stderr | stdout | file=<filename> | <filename> */
+				if (0 == j9_cmdla_stricmp("stderr", output)) {
+					if (NULL != upToColon) {
+						goto xlogerr;
+					}
+				} else if (0 == j9_cmdla_stricmp("stdout", output)) {
+					if (NULL != upToColon) {
+						goto xlogerr;
+					}
+					/* Can't log to stdout yet */
+					unrecognizedOption = TRUE;
+				} else if (NULL == upToColon) {
+					if ('\0' != *output) {
+						/* Without `=` anything non-empty is a filename */
+						gclogfile = output;
+					}
+				} else if (0 == j9_cmdla_stricmp("file", output)) {
+					if (NULL == upToColon) {
+						/* `file=` with no filename specified */
+						goto xlogerr;
+					}
+					gclogfile = upToColon;
+				} else {
+					/* Anything else such as `<something>=<something>` is an error */
+					goto xlogerr;
+				}
+
+				if (NULL != fullOptions) {
+					/* Anything besides trailing ':'s is unrecognized */
+					if (('\0' != *fullOptions) && (':' != *fullOptions)) {
+						unrecognizedOption = TRUE;
+					} else {
+						strUpToDelimiter(&fullOptions, ':');
+						if ((NULL != fullOptions) && ('\0' != *fullOptions)) {
+							unrecognizedOption = TRUE;
+						}
+					}
+				}
+			}
+		}
+#if defined(DEBUG_XLOG)
+		j9tty_printf(PORTLIB, "setgclog %d gclog %d file %s\n", setgclog, gclog, gclogfile);
+#endif /* defined(DEBUG_XLOG) */
+		if (setgclog && gclog) {
+		 	if (NULL != gclogfile) {
```

Comment:
nit: strange indentation (space between tabs)

Reply:
Fixed (and squashed and rebased).